app/src/main/java/com/babylon/wallet/android/presentation/ui/composables/ActionableAddressView.kt

```diff
@@ -71,24 +74,66 @@ fun ActionableAddressView(
-        mutableStateOf(
-            resolveStaticActions(
-                actionableAddress = actionableAddress,
-                context = context
-            )
-        )
+    val scope = rememberCoroutineScope()
+
+    var actions: PopupActions? by remember(actionableAddress) {
+        mutableStateOf(null)
     }
 
-    // Resolve if address is ledger and attach another action
-    if (actionableAddress.type == ActionableAddress.Type.ACCOUNT) {
-        val useCaseProvider = remember(context) {
-            EntryPoints.get(context.applicationContext, ActionableAddressViewEntryPoint::class.java)
-        }
-        val scope = rememberCoroutineScope()
+    val useCaseProvider = remember(context) {
+        EntryPoints.get(context.applicationContext, ActionableAddressViewEntryPoint::class.java)
+    }
+
+    LaunchedEffect(actionableAddress) {
+        scope.launch {
+            val networkId = useCaseProvider.currentGatewayUseCase().invoke().network.networkId()
+
+            val copyAction = PopupActionItem(
+                name = context.getString(
+                    when {
+                        actionableAddress.type == ActionableAddress.Type.TRANSACTION -> R.string.addressAction_copyTransactionId
+                        actionableAddress.isNft -> R.string.addressAction_copyNftId
+                        else -> R.string.addressAction_copyAddress
+                    }
+                ),
+                icon = R.drawable.ic_copy
+            ) { OnAction.CallbackBasedAction.CopyToClipboard(actionableAddress) }
+
+            val openExternalAction = PopupActionItem(
+                name = context.getString(R.string.addressAction_viewOnDashboard),
+                icon = R.drawable.ic_external_link,
+            ) {
+                OnAction.CallbackBasedAction.OpenExternalWebView(
+                    actionableAddress = actionableAddress,
+                    networkId = networkId
+                )
+            }
+
+            val qrAction = PopupActionItem(
+                name = context.getString(R.string.addressAction_showAccountQR),
+                icon = com.babylon.wallet.android.designsystem.R.drawable.ic_qr_code_scanner
+            ) { OnAction.ViewBasedAction.QRCode(actionableAddress) }
 
-        LaunchedEffect(actionableAddress) {
-            scope.launch {
+            actions = if (actionableAddress.isCopyPrimaryAction) {
+                val secondaryActions = if (actionableAddress.type == ActionableAddress.Type.ACCOUNT) {
+                    listOf(qrAction, openExternalAction)
+                } else {
+                    listOf(openExternalAction)
+                }
+
+                PopupActions(
+                    primary = copyAction,
+                    secondary = secondaryActions
+                )
+            } else {
+                PopupActions(
+                    primary = openExternalAction,
+                    secondary = listOf(copyAction)
+                )
+            }
```

Comment:
This can go last, after we know the ledger action too so 

```
... [copyAction, openExternalAction, qrAction]...
val ledgerVerificationAction: PopupActionItem? = if (...) .. else null

actions = if (actionableAddress.isCopyPrimaryAction) {
                val secondaryActions = if (actionableAddress.type == ActionableAddress.Type.ACCOUNT) {
                    listOf(qrAction, openExternalAction, ledgerVerificationAction)
                } else {
                    listOf(openExternalAction, ledgerVerificationAction)
                }

                PopupActions(
                    primary = copyAction,
                    secondary = secondaryActions
                )
            } else {
                PopupActions(
                    primary = openExternalAction,
                    secondary = listOf(copyAction)
                )
            }
```

Reply:
The logic seems to be slightly different than this simplification as far as I can see and you are mixing all actions all together instead of separating them into primary and secondary so I will leave it as is for now.